syntax = "proto3";
package zkp_auth;

message RegisterRequest {
    string user = 1;
    int64 y1 = 2;
    int64 y2 = 3;
}

message RegisterResponse {
    //whether or not the message was successfully registered
    //changed from original template, which had this message empty (no fields)
    string result = 1; 
}

message AuthenticationChallengeRequest {
    string user = 1;
    int64 r1 = 2;
    int64 r2 = 3;
}

message AuthenticationChallengeResponse {
    string auth_id = 1;
    int64 c = 2;
}

message AuthenticationAnswerRequest {
    string auth_id = 1;
    int64 s = 2;
}

message AuthenticationAnswerResponse {
    string session_id = 1;
}

service Auth {
    rpc Register(RegisterRequest) returns(RegisterResponse){}
    rpc CreateAuthenticationChallenge(AuthenticationChallengeRequest) returns(AuthenticationChallengeResponse){}
    rpc VerifyAuthentication(AuthenticationAnswerRequest) returns(AuthenticationAnswerResponse){}
}